### ISSUE TITLE
Can't run llama model on Intel GPU on Linux platforms

### ISSUE THREAD

**Issue by acane77** [intel-analytics/ipex-llm#12222]
**ipex version**: ipex-llm==2.2.0b20241010
**System**: Ubuntu 22.04
**Intel OneAPI 2024.2**

**Description**

IPEX cannot infer with multi-round chat (i.e. prefill a new prompt without clearing the KV cache)

When we tried a multi-round chat the library crashes.

The error message is as following:

```
llm_chat_cli: /home/runner/_work/llm.cpp/llm.cpp/llm.cpp/bigdl-core-xe/llama_backend/sdp_kernel.cpp:639: void sdp_fp8_casual_kernel(const void *, const uint8_t *, const uint8_t *, void *, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, const size_t, float *, float, sycl::queue &) [GS = 32, HD = 96]: Assertion `(context_length-seq_len)%GS==0 && "ubatch must be set as the times of GS\n"' failed.


```

Also, it works well on Windows system, everything is just fine without modofying any codes.

**Comment by acane77** [comment]
We tried `ipex-llm[cpp]==2.2.0b20240928`, no problems found and no crashes.

Also, the Windows version mentioned above is also this `2.2.0b20240928` version.

**Comment by qiuxin2012** [comment]
We have reproduced your issue and are currently working on a fix.

**Comment by qiuxin2012** [comment]
It's fixed now，you can update ipex-llm[cpp] to 2.2.0b20241021 and try again. @acane77

**Comment by user7z** [comment]
@acane77 @qiuxin2012 also ipex-llm works with 2024.0 in linux  & 2024.2 for windows

**Comment by acane77** [comment]
> It's fixed now，you can update ipex-llm[cpp] to 2.2.0b20241021 and try again. @acane77

We've tried this version and it fixes, thanks
